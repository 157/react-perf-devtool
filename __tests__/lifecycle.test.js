var MEASURES = require('../samples/measures')

var { getLifecycleTime, getTotalMethods } = require('../src/shared/lifecycle')
var { getTotalTime } = require('../src/shared/totalTime')

describe('Calling Lifecycle Methods', () => {
  test('Sanity', () => {
    expect(getLifecycleTime(MEASURES)).toMatchSnapshot()
  })

  test('Total time taken in calling lifecycle methods', () => {
    expect(getTotalTime(getLifecycleTime(MEASURES))).toMatchSnapshot()
  })

  test('Total effects', () => {
    expect(getTotalMethods(getLifecycleTime(MEASURES))).toMatchSnapshot()
  })
})
