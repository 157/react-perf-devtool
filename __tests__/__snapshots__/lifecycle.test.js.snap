// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calling Lifecycle Methods Sanity 1`] = `
Object {
  "Calling Lifecycle Methods": Object {
    "timeSpent": Array [
      4.38,
      0.89,
    ],
    "totalMethods": Array [
      6,
      1,
    ],
  },
}
`;

exports[`Calling Lifecycle Methods Total effects 1`] = `7`;

exports[`Calling Lifecycle Methods Total time taken in calling lifecycle methods 1`] = `5.27`;
